Add szr35 keyboard

diegorodriguezv opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->
Add support for SZRKBD SZR35 keyboard.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

diegorodriguezv committed:
Initial szr35 support

diegorodriguezv committed:
Update readme.md

Add images hosted on github.

diegorodriguezv committed:
Updated factory keymap to qmk.

diegorodriguezv committed:
Added instructions to flash the factory keymap.

waffle87 commented on keyboards/szrkbd/szr35/config.h:
```diff
@@ -0,0 +1,17 @@
+
+#pragma once
+
+#define TAPPING_TOGGLE 2
```
This option does not appear to be used anywhere. Should be left to the end user to configure anyway.
```suggestion
```

waffle87 commented on keyboards/szrkbd/szr35/config.h:
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#define TAPPING_TOGGLE 2
+#define DYNAMIC_KEYMAP_LAYER_COUNT 9
```
```suggestion
```

waffle87 commented on keyboards/szrkbd/szr35/keyboard.json:
```diff
@@ -0,0 +1,219 @@
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true
+    },
```
```suggestion
    },
    "dynamic_keymap": {
        "layer_count": 9
    },
```

waffle87 commented on keyboards/szrkbd/szr35/config.h:
```diff
@@ -0,0 +1,17 @@
+
+#define WEAR_LEVELING_LOGICAL_SIZE 2048
+#define WEAR_LEVELING_BACKING_SIZE (WEAR_LEVELING_LOGICAL_SIZE * 2)
+
+#define USB_POLLING_INTERVAL_MS 1
```
Default behaviour.
```suggestion
```

waffle87 commented on keyboards/szrkbd/szr35/keyboard.json:
```diff
@@ -0,0 +1,219 @@
+            "watchdog_timeout": 3000
+        },
+        "usb_detect": {
+            "enabled": true,
```
Default behaviour.
```suggestion
```

waffle87 commented on keyboards/szrkbd/szr35/keyboard.json:
```diff
@@ -0,0 +1,219 @@
+        "bootmagic": {
+            "matrix": [4, 4]
+        }
+    },
```
```suggestion
    },
    "community_layouts": ["split_3x5_3"],
```

waffle87 commented on keyboards/szrkbd/szr35/keyboard.json:
```diff
@@ -0,0 +1,219 @@
+    },
+    "rgb_matrix": {
+        "driver": "ws2812",
+        "led_count": 36,
+        "split": true,
```
```suggestion
```

waffle87 commented on keyboards/szrkbd/szr35/keyboard.json:
```diff
@@ -0,0 +1,219 @@
+        "split": true,
+        "split_count": [18, 18],
+        "animations": {
+            "solid_color": true,
```
```suggestion
```

waffle87 commented on keyboards/szrkbd/szr35/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,67 @@
+// switch easily between the 36-key keyboard and a laptop keyboard. The goal is to minimize the
+// learning curve by avoiding key relocation and maintaining our hard-earned muscle memory.
+
+#include "keycodes.h"
+#include "keymap_us.h"
```
```suggestion
```

waffle87 commented on keyboards/szrkbd/szr35/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,67 @@
+#define G_TAB G(KC_TAB)
+#define C_ESC C(KC_ESC)
+
+#define ENT_LY1 LT(LAYER1, KC_ENT)
+#define BSP_LY2 LT(LAYER2, KC_BSPC)
```
```suggestion
#define ENT_LY1 LT(_LAYER_1, KC_ENT)
#define BSP_LY2 LT(_LAYER_2, KC_BSPC)
```

waffle87 commented on keyboards/szrkbd/szr35/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,67 @@
+    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,        KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,
+    KC_F11,  KC_F12,  KC_MPRV, KC_MPLY, KC_MNXT,      MS_LEFT, MS_DOWN, MS_UP,   MS_RGHT, KC_PSCR,
+    KC_MUTE, KC_VOLU, KC_VOLD, MS_WHLD, MS_WHLU,      MS_BTN1, MS_BTN2, MS_BTN3, MS_BTN4, MS_BTN5,
+                      MS_ACL0, MS_ACL1, MS_ACL2,      XXXXXXX, LAYER3,  XXXXXXX
```
```suggestion
                      MS_ACL0, MS_ACL1, MS_ACL2,      XXXXXXX, LAYER3,  _______
```

diegorodriguezv committed:
Update readme according to the template.

diegorodriguezv committed:
Invert volume keys.

diegorodriguezv committed:
Fixes graciously suggested by @waffle87

diegorodriguezv commented:
Thank you very much for your suggestions @waffle87!
Any other comments would be appreciated.

diegorodriguezv committed:
Kind suggestions by @zvecr

diegorodriguezv committed:
Update keyboards/szrkbd/szr35/readme.md

Co-authored-by: Joel Challis <git@zvecr.com>

diegorodriguezv committed:
Update keyboards/szrkbd/szr35/readme.md

Co-authored-by: Joel Challis <git@zvecr.com>

diegorodriguezv committed:
Update keyboards/szrkbd/szr35/readme.md

Co-authored-by: Joel Challis <git@zvecr.com>

diegorodriguezv committed:
Update keyboards/szrkbd/szr35/readme.md

Co-authored-by: Joel Challis <git@zvecr.com>

diegorodriguezv committed:
Fix typo.

diegorodriguezv commented:
Thank you very much @zvecr!

diegorodriguezv committed:
Removed a couple of extra spaces.

waffle87 commented:
There are a couple lingering suggestions that have not been addressed.

zvecr commented:
> I didn't use the web GitHub interface. I made the changes in my computer

And by doing so, you have not correctly implemented the suggestion.

diegorodriguezv committed:
Update keyboards/szrkbd/szr35/keymaps/factory/keymap.c

Co-authored-by: Joel Challis <git@zvecr.com>

diegorodriguezv commented:
My apologies. Thanks for pointing out the problem.

diegorodriguezv committed:
Add delete to the default keymap

It commonly deletes items in graphical applications.

zvecr commented on keyboards/szrkbd/szr35/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,67 @@
+// switch easily between the 36-key keyboard and a laptop keyboard. The goal is to minimize the
+// learning curve by avoiding key relocation and maintaining our hard-earned muscle memory.
+
+#include "keycodes.h"
```
This shouldn't be added back
```suggestion
```

diegorodriguezv committed:
Update keyboards/szrkbd/szr35/keymaps/default/keymap.c

Co-authored-by: Joel Challis <git@zvecr.com>

saurabswaarm commented:
Hey Diego, is this PR good to use, sorry very new to QMK and don't want to brick the board. 

Does it provide VIAL support ?

And I believe there was as bug in the RGB modes inthe factory firmware, I managed to crash my board while cycling through the RGBs. Had to reset.

Thankyou so much to Noel and you for helping everyone out !! Really helps us out, as this board is very affordable in my country !!

diegorodriguezv commented:
Hi @saurabswaarm.

This is totally usable, it fixes several bugs with the factory firmware.

You must understand that qmk and vial are different projects. Vial builds on top of qmk. I'm waiting for this PR to be accepted so that I can commit the vial support on that project. In the meantime you can use my vial fork at: https://github.com/diegorodriguezv/vial-qmk

More info about backing up your firmware and the flashing process can be found at my other project: https://github.com/diegorodriguezv/szr35

Hope it helps.

diegorodriguezv commented:
Hi guys. Is there anything I can do to speed up the approval of this PR?

diegorodriguezv committed:
Update keyboards/szrkbd/szr35/keyboard.json

Removed trailing comma

Co-authored-by: フィルターペーパー <76888457+filterpaper@users.noreply.github.com>

diegorodriguezv commented:
Hello.

Is there something else needed to merge this? 
How can I speed up the review process? 
Did I miss any requested change?

Thanks.

drashna commented on keyboards/szrkbd/szr35/keyboard.json:
```diff
@@ -0,0 +1,221 @@
+    "processor": "STM32F401",
+    "usb": {
+        "pid": "0x45D4",
+        "vid": "0x3601",
```
This vendor ID is in use, by [VALEO EMBRAYAGES](https://www.the-sz.com/products/usbid/index.php?v=0x3601).

The further past 0x4000, the better, as the 0x3xxx block is being actively assigned

diegorodriguezv replied:
Thanks for catching that. It was part of the factory firmware. Can I just choose any unused one?

drashna replied:
There are a couple ways to get a legit vid+pid, such as pid.codes, Microchip, etc.  But generally, yeah.   The higher the better, and you can see a list of used values in the repo here: 
https://yanfali.github.io/qmk_usb_usage/

diegorodriguezv replied:
Thanks. Changed it to: `0x44DF`. I checked both links and it's not used.

diegorodriguezv committed:
Changed the vendor ID

@drashna astutely pointed out the previous vendor ID was used. Changed
it for an unused one.
